### ISSUE TITLE
Request: add a --tag flag (or something similar)

### ISSUE THREAD

**Issue by gswallow** [sparkleformation/sfn#16]
Hi,

In https://github.com/miasma-rb/miasma-aws/blob/master/lib/miasma/contrib/aws/orchestration.rb, there's a way to add tags to a stack, itself, but I don't think there are any ways to tag a stack in the stack's json file.  It would be nice to be able to add an "environment" tag to a stack while I create it with 'sfn create,' so that I can easily identify stacks for teardown and cleanup after testing.

Thanks as usual!

**Comment by chrisroberts** [comment]
Hi! I'm going to leave this open as an enhancement since there is currently no CLI flag to provide tags. You can provide tags via the config:

```ruby
Configuration.new do
  options do
    tags do
      environment ENV.fetch('SFN_ENVIRONMENT', 'testing')
    end
  end
end
```

Not quite as flexible as the CLI flags but it at least provides access to the feature right now :smiley:

**Comment by luckymike** [comment]
@gswallow you're correct that the CFN API does not support adding tags via the template. :+1: to a `--tags` flag!

**Comment by chrisroberts** [comment]
One other note is that tags do not propagate, so nested stacks will not end up with tags set on the parent. This will hopefully be addressed in the next release via before/after callback support which will find all ancestors of the root stack and automatically sync tags.

**Comment by gswallow** [comment]
Adding the environment to the .sfn file works fine.  Thanks!

**Comment by mfischer-zd** [comment]
Any update on this?

**Comment by kabakakao** [comment]
:+1: to a --tags flag!

**Comment by autarchprinceps** [comment]
Given that resource groups in AWS are tagged based, among many other things, this would be really important to have, especially if combined with tag support in sfn-parameters. Since you do support setting tags on CloudFormation Stacks at all, shouldn't this be just internally adding the option where to get that information from?

**Comment by autarchprinceps** [comment]
I created working code that adds a --tags option, that works like --parameters and merges with .sfn option for parameters.
